src/views/pacbio/PacbioLibraryBatchCreate.vue

```diff
@@ -0,0 +1,302 @@
+ *
+ */
+const setDefaultTagSet = () => {
+  const defaultTagSet = pacbioRootStore.tagSetChoices.find(
+    (option) => option.text === defaultTagSetName,
+  )
+  if (defaultTagSet) {
+    selectedTagSet.value = defaultTagSet.value
+  }
+}
+
+async function fetchData() {
+  await pacbioRootStore.fetchPacbioTagSets()
+  setDefaultTagSet()
+  await fetchPrinters()
+  return { success: true }
+}
+
+const isDisabledCSVPreview = computed(() => !state.csvData.length)
+
+const isCreateDisabled = computed(() => !selectedTagSet.value || !selectedCSVFile.value)
+
+const printBarcodes = computed(() => state.resultData.map((item) => item.barcode))
+
+const isPrintDisabled = computed(() => !selectedPrinterName.value || !printBarcodes.value.length)
+
+/**
+ * Printer options
+ * @returns {Array} - Printer options
+ * @example [{ value: 1, text: 'Printer 1' }, { value: 2, text: 'Printer 2' }]
+ */
+// printerOptions is used to display the printers that are available to print to
+const printerOptions = computed(() => {
+  return printers('tube').map(({ name }) => ({
+    text: name,
+  }))
+})
+
+const selectedCSVFile = ref('') //Reference to the selected csv file
+const csvTableFields = [
+  { key: 'source', label: 'Source' },
+  { key: 'tag', label: 'Tag' },
+  { key: 'template_prep_kit_box_barcode', label: 'Template prep kit box barcode' },
+  { key: 'volume', label: 'Volume' },
+  { key: 'concentration', label: 'Concentration' },
+  { key: 'insert_size', label: 'Insert size' },
+]
+//define reactive variables
+const state = reactive({
+  csvTableFields,
+  // Define fields for the table
+  resultTableFields: [{ key: 'barcode', label: 'Barcode' }, ...csvTableFields],
+  csvData: [],
+  resultData: [],
+})
+
+/**
+ * Handles the csv file selection
+ * @param evt - Event object
+ */
+const onSelectFile = async (evt) => {
+  if (evt?.target?.files?.length) {
+    selectedCSVFile.value = evt.target.files[0]
+    try {
+      const csv = await selectedCSVFile.value.text()
+      //Parse the csv file
+      const records = eachRecord(csv, () => {})
+      state.csvData = records.map((record) => record.record)
+    } catch (error) {
+      showAlert(error, 'danger')
+    }
+  }
+}
+
+/**
+ * Resets the form
+ */
+const onReset = () => {
+  state.csvData = []
+  state.resultData = []
+  csvFileInput.value.value = ''
+  selectedPrinterName.value = null
+  setDefaultTagSet()
+}
+
+/**
+ * Creates a library batch from the selected tag set and csv file
+ */
+const createLibraryBatch = async () => {
+  const selectedTagSetName = pacbioRootStore.tagSetChoices.find(
+    (tagset) => tagset.value === selectedTagSet.value,
+  )?.text
+  if (!selectedTagSetName) {
+    showAlert('Please select a tag set', 'danger')
+    return
+  }
+  const { success, errors, result } = await pacbioLibraryBatchCreateStore.createLibraryBatch(
+    selectedCSVFile.value,
+    selectedTagSetName,
+  )
+  if (success) {
+    state.resultData = result
+  } else {
+    showAlert(errors, 'danger')
```

Comment:
showAlert should take a string as an error not an array, its throwing a warning in the console. Might be worthing adding a join(',') to errors.

Reply:
I think addMessage in rootStore is expecting errors as an object. So, I will change the store where it is returned as an array